### ISSUE TITLE
`destructuring-assignment` requires destructuring for language keywords

### ISSUE THREAD

**Issue by constgen** [yannickcr/eslint-plugin-react#3027]
```jsx
<Fade in={this.props.in} />
```
`destructuring-assignment` rule finds a violation here and requires the destructuring like this

```js
let { in } = this.props
```
But this throws a Syntax Error. We need a way to manually exclude some names from the rule or exclude all JavaScript keywords by default, e.g. `if`, `while`, `in`, `for`, others.

**Comment by ljharb** [comment]
Nope, this should still be destructured - you'll just need to rename it:
```jsx
const { in: inProp } = this.props
```

**Comment by constgen** [comment]
You convinced me
